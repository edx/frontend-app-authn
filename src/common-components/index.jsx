export { default as HeaderLayout } from './HeaderLayout';
export { default as RedirectAuthn } from './RedirectAuthn';
export { default as registerIcons } from './RegisterFaIcons';
export { default as UnAuthOnlyRoute } from './UnAuthOnlyRoute';
export { default as NotFoundPage } from './NotFoundPage';
