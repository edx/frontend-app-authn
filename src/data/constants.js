// URL Paths
export const LOGIN_PAGE = '/login';
export const REGISTER_PAGE = '/register';
export const RESET_PAGE = '/reset';
export const DEFAULT_REDIRECT_URL = '/dashboard';
export const PASSWORD_RESET_CONFIRM = '/password_reset_confirm/:token/';
export const PAGE_NOT_FOUND = '/notfound';

// Stateful Submit Button States
export const DEFAULT_STATE = 'default';
export const PENDING_STATE = 'pending';

// Constants
export const SUPPORTED_ICON_CLASSES = ['apple', 'facebook', 'google', 'microsoft'];
export const REGISTRATION_VALIDITY_MAP = {};
export const REGISTRATION_OPTIONAL_MAP = {};
export const REGISTRATION_EXTRA_FIELDS = [
  'confirm_email',
  'level_of_education',
  'gender',
  'year_of_birth',
  'mailing_address',
  'goals',
  'honor_code',
  'terms_of_service',
  'city',
  'country',
];

export const VALID_EMAIL_REGEX = '(^[-!#$%&\'*+/=?^_`{}|~0-9A-Z]+(\\.[-!#$%&\'*+/=?^_`{}|~0-9A-Z]+)*'
                                 + '|^"([\\001-\\010\\013\\014\\016-\\037!#-\\[\\]-\\177]|\\\\[\\001-\\011\\013\\014\\016-\\177])*"'
                                 + ')@((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\\.)+)(?:[A-Z0-9-]{2,63})'
                                 + '|\\[(25[0-5]|2[0-4]\\d|[0-1]?\\d?\\d)(\\.(25[0-5]|2[0-4]\\d|[0-1]?\\d?\\d)){3}\\]$';
