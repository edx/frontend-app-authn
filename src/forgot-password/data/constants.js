export default 'internal-server-error';
