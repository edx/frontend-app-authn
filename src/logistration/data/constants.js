export const INACTIVE_USER = 'inactive-user';
export const INTERNAL_SERVER_ERROR = 'internal-server-error';
export const NON_COMPLIANT_PASSWORD_EXCEPTION = 'NonCompliantPasswordException';
