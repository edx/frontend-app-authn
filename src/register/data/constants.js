export const YEAR_OF_BIRTH_OPTIONS = (() => {
  const currentYear = new Date().getFullYear();
  const years = [];
  let startYear = currentYear - 120;
  while (startYear < currentYear) {
    startYear += 1;

    years.push({ value: startYear.toString(), label: startYear });
  }
  return years.reverse();
})();

export const EDUCATION_LEVELS = [
  '',
  'p',
  'm',
  'b',
  'a',
  'hs',
  'jhs',
  'el',
  'none',
  'o',
];

export const GENDER_OPTIONS = ['', 'f', 'm', 'o'];
