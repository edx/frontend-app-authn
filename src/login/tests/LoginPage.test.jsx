import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import { getConfig } from '@edx/frontend-platform';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';
import * as analytics from '@edx/frontend-platform/analytics';
import LoginPage from '../LoginPage';
import { RenderInstitutionButton } from '../../common-components';
import { PENDING_STATE } from '../../data/constants';

jest.mock('@edx/frontend-platform/analytics');

analytics.sendTrackEvent = jest.fn();
analytics.sendPageEvent = jest.fn();

const IntlLoginPage = injectIntl(LoginPage);
const mockStore = configureStore();

describe('LoginPage', () => {
  const initialState = {
    forgotPassword: { status: null },
    login: {
      loginResult: { success: false, redirectUrl: '' },
    },
    commonComponents: {
      thirdPartyAuthApiStatus: null,
      thirdPartyAuthContext: {
        currentProvider: null,
        finishAuthUrl: null,
        providers: [],
        secondaryProviders: [],
      },
    },
  };

  let props = {};
  let store = {};

  const secondaryProviders = {
    id: 'saml-test',
    name: 'Test University',
    loginUrl: '/dummy-auth',
    registerUrl: '/dummy_auth',
  };

  const appleProvider = {
    id: 'oa2-apple-id',
    name: 'Apple',
    iconClass: null,
    iconImage: 'https://edx.devstack.lms/logo.png',
    loginUrl: '/auth/login/apple-id/?auth_entry=login&next=/dashboard',
  };

  const reduxWrapper = children => (
    <IntlProvider locale="en">
      <Provider store={store}>{children}</Provider>
    </IntlProvider>
  );

  beforeEach(() => {
    store = mockStore(initialState);
    props = {
      loginRequest: jest.fn(),
    };
  });

  it('should match default section snapshot', () => {
    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match pending button state snapshot', () => {
    store = mockStore({
      ...initialState,
      login: {
        ...initialState.login,
        submitState: PENDING_STATE,
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match forget password alert message snapshot', () => {
    store = mockStore({
      ...initialState,
      forgotPassword: { status: 'complete', email: 'test@example.com' },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match TPA provider snapshot', () => {
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          providers: [appleProvider],
        },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show error message', () => {
    store = mockStore({
      ...initialState,
      login: {
        ...initialState.login,
        loginError: { value: 'Email or password is incorrect.' },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show account activation message', () => {
    delete window.location;
    window.location = { href: getConfig().BASE_URL.concat('/login?account_activation_status=info') };

    const expectedMessage = 'This account has already been activated.';

    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(loginPage.find('#account-activation-message').find('div').text()).toEqual(expectedMessage);
  });

  it('should display login help button', () => {
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.find('button.field-link').first().text()).toEqual('Need help signing in?');
  });

  it('updates the error state for invalid email', () => {
    const errorState = { email: null, password: '' };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('input#loginPassword').simulate('change', { target: { value: 'test', name: 'password' } });
    loginPage.find('button.btn-brand').simulate('click');

    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('updates the error state for invalid password', () => {
    const errorState = { email: '', password: null };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('input#loginEmail').simulate('change', { target: { value: 'test@example.com', name: 'email' } });
    loginPage.find('button.btn-brand').simulate('click');
    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('should match url after redirection', () => {
    const dasboardUrl = 'http://test.com/testing-dashboard/';
    store = mockStore({
      ...initialState,
      login: {
        ...initialState.login,
        loginResult: {
          success: true,
          redirectUrl: dasboardUrl,
        },
      },
    });
    delete window.location;
    window.location = { href: getConfig().BASE_URL };
    renderer.create(reduxWrapper(<IntlLoginPage {...props} />));
    expect(window.location.href).toBe(dasboardUrl);
  });

  it('should match url after TPA redirection', () => {
    const authCompleteUrl = '/auth/complete/google-oauth2/';
    store = mockStore({
      ...initialState,
      login: {
        ...initialState.login,
        loginResult: {
          success: true,
          redirectUrl: '',
        },
      },
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          finishAuthUrl: authCompleteUrl,
        },
      },
    });

    delete window.location;
    window.location = { href: getConfig().BASE_URL };
    renderer.create(reduxWrapper(<IntlLoginPage {...props} />));
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + authCompleteUrl);
  });

  it('should redirect to social auth provider url', () => {
    const loginUrl = '/auth/login/apple-id/?auth_entry=login&next=/dashboard';
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          providers: [{
            ...appleProvider,
            loginUrl,
          }],
        },
      },
    });

    delete window.location;
    window.location = { href: getConfig().BASE_URL };

    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('button#oa2-apple-id').simulate('click');
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + loginUrl);
  });

  it('should match third party auth alert', () => {
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          currentProvider: 'Apple',
          platformName: 'edX',
        },
      },
    });

    const expectedMessage = 'You have successfully signed into Apple, but your Apple account does not have a '
                            + 'linked edX account. To link your accounts, sign in now using your edX password.';

    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(loginPage.find('#tpa-alert').find('span').text()).toEqual(expectedMessage);
  });

  it('should display institution login button', () => {
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          secondaryProviders: [secondaryProviders],
        },
      },
    });
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.text().includes('Use my university info')).toBe(true);
  });

  it('should not display institution login button', () => {
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.text().includes('Use my university info')).toBe(false);
  });

  it('should display institution login page', () => {
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          secondaryProviders: [secondaryProviders],
        },
      },
    });
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    loginPage.find(RenderInstitutionButton).simulate('click', { institutionLogin: true });
    expect(loginPage.text().includes('Test University')).toBe(true);
  });

  it('send tracking event when create account link is clicked', () => {
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('a[href*="/register"]').simulate('click');
    loginPage.update();
    expect(analytics.sendTrackEvent).toHaveBeenCalledWith('edx.bi.register_form.toggled', { category: 'user-engagement' });
  });

  it('send page event when login page is rendered', () => {
    mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(analytics.sendPageEvent).toHaveBeenCalledWith('login_and_registration', 'login');
  });

  it('send tracking and page events when institutional button is clicked', () => {
    store = mockStore({
      ...initialState,
      commonComponents: {
        ...initialState.commonComponents,
        thirdPartyAuthContext: {
          ...initialState.commonComponents.thirdPartyAuthContext,
          secondaryProviders: [secondaryProviders],
        },
      },
    });
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    loginPage.find(RenderInstitutionButton).simulate('click', { institutionLogin: true });
    expect(analytics.sendTrackEvent).toHaveBeenCalledWith('edx.bi.institution_login_form.toggled', { category: 'user-engagement' });
    expect(analytics.sendPageEvent).toHaveBeenCalledWith('login_and_registration', 'institution_login');
  });
});
