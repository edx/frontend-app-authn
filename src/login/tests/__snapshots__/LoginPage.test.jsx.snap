// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match TPA provider snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="mw-500"
    >
      <div
        className="d-flex flex-row"
      >
        <p>
          First time here?
          <a
            className="ml-1"
            href="/register"
            onClick={[Function]}
            target="_self"
          >
            Create an Account
            .
          </a>
        </p>
      </div>
      <hr
        className="mt-0 border-gray-200"
      />
      <h3
        className="text-left mt-2 mb-3"
      >
        Sign In
      </h3>
      <form
        className="m-0"
      >
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginEmail"
          >
            Email
          </label>
          <input
            aria-describedby="email-help-text"
            className="form-control"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="username@domain.com"
            type="email"
            value=""
          />
          <small
            className="form-text text-muted"
            id="email-help-text"
          >
            The email address you used to register with edX.
          </small>
          <strong
            className="invalid-feedback"
            id="email-invalid-feedback"
          >
            The email address you've provided isn't formatted correctly.
          </strong>
        </div>
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginPassword"
          >
            Password
          </label>
          <input
            aria-describedby=""
            className="form-control"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <strong
            className="invalid-feedback"
            id="password-invalid-feedback"
          >
            Please enter your password.
          </strong>
        </div>
        <button
          className="mt-1 field-link small"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
        <a
          className="field-link mt-0 mb-3 small"
          href="http://localhost:18000/enterprise/login"
          onClick={[Function]}
          target="_self"
        >
          Sign in with your company or school
        </a>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-brand"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            Sign in
          </span>
        </button>
      </form>
      <div
        className="mb-20"
      >
        <hr
          className="mt-20 mb-20 border-gray-200"
        />
        or sign in with
      </div>
      <div
        className="row tpa-container"
      >
        <button
          className="btn-social btn-oa2-apple-id"
          data-provider-url="/auth/login/apple-id/?auth_entry=login&next=/dashboard"
          id="oa2-apple-id"
          onClick={[Function]}
          type="button"
        >
          <div
            aria-hidden="true"
            className="mx-auto"
          >
            <img
              alt="icon Apple"
              className="icon-image"
              src="https://edx.devstack.lms/logo.png"
            />
            <span
              aria-hidden="true"
              className="pl-2"
            >
              Apple
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`LoginPage should match default section snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="mw-500"
    >
      <div
        className="d-flex flex-row"
      >
        <p>
          First time here?
          <a
            className="ml-1"
            href="/register"
            onClick={[Function]}
            target="_self"
          >
            Create an Account
            .
          </a>
        </p>
      </div>
      <hr
        className="mt-0 border-gray-200"
      />
      <h3
        className="text-left mt-2 mb-3"
      >
        Sign In
      </h3>
      <form
        className="m-0"
      >
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginEmail"
          >
            Email
          </label>
          <input
            aria-describedby="email-help-text"
            className="form-control"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="username@domain.com"
            type="email"
            value=""
          />
          <small
            className="form-text text-muted"
            id="email-help-text"
          >
            The email address you used to register with edX.
          </small>
          <strong
            className="invalid-feedback"
            id="email-invalid-feedback"
          >
            The email address you've provided isn't formatted correctly.
          </strong>
        </div>
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginPassword"
          >
            Password
          </label>
          <input
            aria-describedby=""
            className="form-control"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <strong
            className="invalid-feedback"
            id="password-invalid-feedback"
          >
            Please enter your password.
          </strong>
        </div>
        <button
          className="mt-1 field-link small"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
        <a
          className="field-link mt-0 mb-3 small"
          href="http://localhost:18000/enterprise/login"
          onClick={[Function]}
          target="_self"
        >
          Sign in with your company or school
        </a>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-brand"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            Sign in
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`LoginPage should match forget password alert message snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="mw-500"
    >
      <div
        className="fade alert alert-success show"
        id="confirmation-alert"
        role="alert"
      >
        <div
          className="alert-heading h4"
        >
          Check Your Email
        </div>
        <p>
          <span>
            You entered 
            <strong>
              test@example.com
            </strong>
            . If this email address is associated with your edX account, we will send a message with password recovery instructions to this email address.
          </span>
        </p>
        <p>
          If you do not receive a password reset message after 1 minute, verify that you entered the correct email address, or check your spam folder.
        </p>
        <p>
          <span>
            If you need further assistance, 
            <a
              className="alert-link"
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              contact technical support
            </a>
            .
          </span>
        </p>
      </div>
      <div
        className="d-flex flex-row"
      >
        <p>
          First time here?
          <a
            className="ml-1"
            href="/register"
            onClick={[Function]}
            target="_self"
          >
            Create an Account
            .
          </a>
        </p>
      </div>
      <hr
        className="mt-0 border-gray-200"
      />
      <h3
        className="text-left mt-2 mb-3"
      >
        Sign In
      </h3>
      <form
        className="m-0"
      >
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginEmail"
          >
            Email
          </label>
          <input
            aria-describedby="email-help-text"
            className="form-control"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="username@domain.com"
            type="email"
            value=""
          />
          <small
            className="form-text text-muted"
            id="email-help-text"
          >
            The email address you used to register with edX.
          </small>
          <strong
            className="invalid-feedback"
            id="email-invalid-feedback"
          >
            The email address you've provided isn't formatted correctly.
          </strong>
        </div>
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginPassword"
          >
            Password
          </label>
          <input
            aria-describedby=""
            className="form-control"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <strong
            className="invalid-feedback"
            id="password-invalid-feedback"
          >
            Please enter your password.
          </strong>
        </div>
        <button
          className="mt-1 field-link small"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
        <a
          className="field-link mt-0 mb-3 small"
          href="http://localhost:18000/enterprise/login"
          onClick={[Function]}
          target="_self"
        >
          Sign in with your company or school
        </a>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-brand"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            Sign in
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`LoginPage should match pending button state snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="mw-500"
    >
      <div
        className="d-flex flex-row"
      >
        <p>
          First time here?
          <a
            className="ml-1"
            href="/register"
            onClick={[Function]}
            target="_self"
          >
            Create an Account
            .
          </a>
        </p>
      </div>
      <hr
        className="mt-0 border-gray-200"
      />
      <h3
        className="text-left mt-2 mb-3"
      >
        Sign In
      </h3>
      <form
        className="m-0"
      >
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginEmail"
          >
            Email
          </label>
          <input
            aria-describedby="email-help-text"
            className="form-control"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="username@domain.com"
            type="email"
            value=""
          />
          <small
            className="form-text text-muted"
            id="email-help-text"
          >
            The email address you used to register with edX.
          </small>
          <strong
            className="invalid-feedback"
            id="email-invalid-feedback"
          >
            The email address you've provided isn't formatted correctly.
          </strong>
        </div>
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginPassword"
          >
            Password
          </label>
          <input
            aria-describedby=""
            className="form-control"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <strong
            className="invalid-feedback"
            id="password-invalid-feedback"
          >
            Please enter your password.
          </strong>
        </div>
        <button
          className="mt-1 field-link small"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
        <a
          className="field-link mt-0 mb-3 small"
          href="http://localhost:18000/enterprise/login"
          onClick={[Function]}
          target="_self"
        >
          Sign in with your company or school
        </a>
        <button
          aria-disabled={true}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-pending disabled btn btn-brand"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            <span
              className="pgn__stateful-btn-icon"
            >
              <span
                aria-hidden={true}
                className="icon fa fa-spinner fa-spin"
                id="Icon1"
              />
            </span>
            Sign in
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`LoginPage should show error message 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="mw-500"
    >
      <div
        className="fade alert alert-danger show"
        id="login-failure-alert"
        role="alert"
      >
        <div
          className="alert-heading h4"
        >
          We couldn't sign you in.
        </div>
        <ul>
          <li>
            Email or password is incorrect.
          </li>
        </ul>
      </div>
      <div
        className="d-flex flex-row"
      >
        <p>
          First time here?
          <a
            className="ml-1"
            href="/register"
            onClick={[Function]}
            target="_self"
          >
            Create an Account
            .
          </a>
        </p>
      </div>
      <hr
        className="mt-0 border-gray-200"
      />
      <h3
        className="text-left mt-2 mb-3"
      >
        Sign In
      </h3>
      <form
        className="m-0"
      >
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginEmail"
          >
            Email
          </label>
          <input
            aria-describedby="email-help-text"
            className="form-control"
            id="loginEmail"
            name="email"
            onChange={[Function]}
            placeholder="username@domain.com"
            type="email"
            value=""
          />
          <small
            className="form-text text-muted"
            id="email-help-text"
          >
            The email address you used to register with edX.
          </small>
          <strong
            className="invalid-feedback"
            id="email-invalid-feedback"
          >
            The email address you've provided isn't formatted correctly.
          </strong>
        </div>
        <div
          className="form-group mb-0 w-100"
        >
          <label
            className="h6 mr-1 form-label"
            htmlFor="loginPassword"
          >
            Password
          </label>
          <input
            aria-describedby=""
            className="form-control"
            id="loginPassword"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <strong
            className="invalid-feedback"
            id="password-invalid-feedback"
          >
            Please enter your password.
          </strong>
        </div>
        <button
          className="mt-1 field-link small"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
            data-icon="caret-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          Need help signing in?
        </button>
        <div
          className="pgn-transition-replace-group position-relative"
          style={
            Object {
              "height": null,
            }
          }
        >
          <div
            style={
              Object {
                "padding": ".1px 0",
              }
            }
          />
        </div>
        <a
          className="field-link mt-0 mb-3 small"
          href="http://localhost:18000/enterprise/login"
          onClick={[Function]}
          target="_self"
        >
          Sign in with your company or school
        </a>
        <button
          aria-disabled={false}
          aria-live="assertive"
          className="pgn__stateful-btn pgn__stateful-btn-state-default btn btn-brand"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          <span
            className="d-flex align-items-center justify-content-center"
          >
            Sign in
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;
